app/serializers/search/goods_nomenclature_serializer.rb

```diff
@@ -79,5 +90,57 @@ def validity_start_date
     def validity_end_date
       super&.iso8601
     end
+
+    def ancestor_1_description_indexed
+      ancestors[0].try(:[], :description_indexed)
+    end
+
+    def ancestor_2_description_indexed
+      ancestors[1].try(:[], :description_indexed)
+    end
+
+    def ancestor_3_description_indexed
+      ancestors[2].try(:[], :description_indexed)
+    end
+
+    def ancestor_4_description_indexed
+      ancestors[3].try(:[], :description_indexed)
+    end
+
+    def ancestor_5_description_indexed
+      ancestors[4].try(:[], :description_indexed)
+    end
+
+    def ancestor_6_description_indexed
+      ancestors[5].try(:[], :description_indexed)
+    end
+
+    def ancestor_7_description_indexed
+      ancestors[6].try(:[], :description_indexed)
+    end
+
+    def ancestor_8_description_indexed
+      ancestors[7].try(:[], :description_indexed)
+    end
+
+    def ancestor_9_description_indexed
+      ancestors[8].try(:[], :description_indexed)
+    end
+
+    def ancestor_10_description_indexed
+      ancestors[9].try(:[], :description_indexed)
+    end
+
+    def ancestor_11_description_indexed
+      ancestors[10].try(:[], :description_indexed)
+    end
+
+    def ancestor_12_description_indexed
+      ancestors[11].try(:[], :description_indexed)
+    end
+
+    def ancestor_13_description_indexed
+      ancestors[12].try(:[], :description_indexed)
+    end
```

Comment:
I hate to say it, but this could be a case where
`define_method`
could be beneficial :)

Reply:
The thing is, it doesn't actually add much - in the sense that I have to manually know the largest number of ancestors and statically define them in the index anyway. I'd say this isn't a good use of that. I'd probably use define_method if I dynamically knew something

Reply:
I think it doesn't add much but I can do